### PR TITLE
Issue#115/daily horoscopes

### LINKED ISSUE #115: Daily Horoscope
We've been asked to link to the Daily Horoscope on Collective World from two other sites and to allow it to be updated on a daily basis.

Further, the daily horoscope has a heading for each zodiac sign and we've been asked to link to each zodiac sign. So we would need to either hardcode the zodiac signs and pray the headings are anchored correctly or, parse the ids out of the content retrieved from the API.

Is this a layout for the REST block, or a new block?

### PR DESCRIPTION
**Pull request by jessica-townsend**
### Related Issues
- Closes #115 

### What Was Accomplished
- Created a new layout for the REST block to be used for a daily horoscope post
- Added support for borders and padding to the REST block
- Added a setting for the REST block to show or hide links to individual Zodiac sections
  - Setting only appears if the Daily Horoscope layout is selected

### How It Was Tested
- [x] Locally on parent theme
- [x] Locally on child theme
- [x] Child theme staging (Thought Catalog)

### How To Test
- Horoscope layout looks very similar in the editor and on the frontend
- Hiding the zodiac links using the block setting in the editor works as expected
- Date is separated from the post title and displayed on its own line
- Zodiac links are not shown if headings don't match with a zodiac symbol

### CODE REVIEW

File: blocks/rest/includes/layout/daily-horoscope/class-daily-horoscope.php [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+		return array();
+	}
+
+	/**
+	 * Get SVG Allowed HTML
+	 * 
+	 * @return array
+	 */
+	public static function get_svg_allowed_html(): array {
```
[the remark this answers is not in the document]
Comment (jessica-townsend) [reply blocks/rest/includes/layout/daily-horoscope/class-daily-horoscope.php]: Thanks, I had removed `wp_kses` but I forgot to remove this function

**Comment by douglas-johnson** [comment]
Outputting the links for the individual zodiac signs isn't working on today's horoscope. The author put `<b>` elements inside the headings, so the regex is failing.

I think the third use of `.*` in this regex is too greedy `<h2.*id="(.*)".*>(.*?)<\/h2>`

When someone does ` <h2 class="wp-block-heading" id="aries"><b>Aries</b></h2>`

`<h2.*id="(.*)".*>` will match all the way to `</b>`

We can add another `?` to make it "lazy" so it stops at the end of the h2 like: `<h2.*id="(.*)".*?>(.*?)<\/h2>`

Or a more specific regex `<h2[^>]*id="([^"]*)"[^>]*>(.*)<\/h2>` which uses `[^>]*` to match anything other than closing bracket instead of any character.

Additionally we should run the content of the heading through `wp_strip_all_tags` sometime before `esc_html` so we don't get this:

Screenshot 2023-12-01 at 10 44 31 AM

**Comment by jessica-townsend** [comment]
Thanks @douglas-johnson, I updated the regex and added `wp_strip_all_tags`. This is up on Thought Catalog develop.

**Review by douglas-johnson — APPROVED** [review]
Looks good on the development site. I would like to watch it handle new horoscope posts over the weekend. Let's talk Monday before merging and taking it live.

**Pull request merged**